docs/hack_loki_operator.md

```diff
@@ -0,0 +1,222 @@
+
+  The above command will deploy the operator to your active Kubernetes cluster defined by your local [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/). The operator will be running in the `default` namespace.
+
+* You can confirm that the operator is up and running using:
+
+  ```console
+  kubectl get pods
+  ```
+
+* Now create a LokiStack instance to get the various components of Loki up and running:
+
+  ```console
+  kubectl apply -f hack/lokistack_gateway_dev.yaml
+  ```
+
+  This will create `distributor`, `compactor`, `ingester`, `querier`, `query-frontend` and `lokistack-gateway` components.
+
+  Confirm that all are up and running for `deployments` using:
+
+  ```console  
+  kubectl rollout status deployment/<DEPLOYMENT_NAME>
+  ```  
+
+  where `<DEPLOYMENT_NAME>` is the name of the deployment and can be found using:
+
+  ```console
+  kubectl get deployments
+  ```
+
+  Confirm that all are up and running for `statefulsets` using:
+
+  ```console  
+  kubectl rollout status statefulset/<STATEFULSET_NAME>  
+  ```  
+
+  where `<STATEFULSET_NAME>` is the name of the statefulset and can be found using:
+
+  ```console
+  kubectl get statefulsets
+  ```
+
+  _Note:_  `lokistack-gateway` is an optional component deployed as part of Loki Operator. It provides secure access to Loki's distributor (i.e. for pushing logs) and query-frontend (i.e. for querying logs) via consulting an OAuth/OIDC endpoint for the request subject.
+
+  If you don't want `lokistack-gateway` component then you can skip it by removing the `--with-lokistack-gateway` args from the `controller-manager` deployment:
+
+  ```console
+  kubectl edit deployment/controller-manager
+  ```
+
+  Delete the flag `--with-lokistack-gateway` from the `args` section and save the file. This will update the deployment and now you can create LokiStack instance using:
+
+  ```console
+  kubectl apply -f hack/lokistack_dev.yaml
+  ```
+
+  This will create `distributor`, `compactor`, `ingester`, `querier` and `query-frontend` components only.
+
+### Cleanup
+
+To cleanup deployments of the operator, you can use:
+
+```console
+make undeploy
+```
+
+It will undeploy controller from the configured Kubernetes cluster in ~/.kube/config
+
+## Hacking on Loki Operator on OpenShift
+
+### Requirements
+
+* Install [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) or [Openshift CLI](https://docs.openshift.com/container-platform/latest/cli_reference/openshift_cli/getting-started-cli.html) for communicating with the cluster. The guide below will be using `kubectl` for the same.
+* Create a running OpenShift cluster on AWS.
+* A container registry that you and your OpenShift cluster can reach. We recommend  [quay.io](https://quay.io/signin/).
+* Create an [S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/create-bucket-overview.html) in one of the AWS Regions.
+
+### Installation of Loki Operator
```

Comment:
A lot of those instructions are a replication of the instructions above. For maintenance simplicity of this document please consolidate the instructions so they appear only once, and you just refer to the relevant instructions in the duplicated location.

Reply:
Although both are doing few common steps but since we refer to both (kind and openshift) as separate topics so this looks much cleaner and step-by-step process. Consolidating them and then refering to duplicate section would make bad experience IMO.